test/e2e/storage/testsuites/subpath.go

```diff
@@ -619,7 +620,7 @@ func SubpathTestPod(f *framework.Framework, subpath, volumeType string, source *
 
 func containerIsUnused(container *v1.Container) bool {
 	// mountImage with nil Args or with just "mounttest" as Args does nothing. Leave everything else
-	return container.Image == mountImage && (container.Args == nil || (len(container.Args) == 1 && container.Args[0] == "mounttest"))
+	return container.Image == mountImage && (container.Args == nil && container.Command == nil || (len(container.Args) == 1 && container.Args[0] == "mounttest"))
```

Comment:
Should the check for `Command = nil` be outside of the `Args` check? ie

```
mountImage && Command == nil && (Args check)
```

Reply:
Actually do we need args check at all? Can we just check Command?

Reply:
Nevermind, args check is still needed because you can run mounttest with just args and no command

Reply:
Still have an outstanding question here (the first one)

Reply:
I think in normal case, when command is not nil, container should not have a single argument "mounttest". "mounttest" is already a command. So in another word, the following case should not exist. 
command is set & container has only one arg "mounttest"

If such case happens, in my current logic, the container will be removed. In your proposed approach, the container will be kept. I think the test should fail in both cases and we should fix the test case. So either way is fine for me. Or we could add logic to check this situation and fail early with some logs indicating the problem. I think this way will help to discover the problem easier.

Reply:
never mind. I think if command is set, it is possible to have any arbitrary args.